From 0ef95f81720c0ea623dd5ecb88c21ad8cad48710 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:39 -0400
Subject: [cpufreq] acpi-cpufreq: don't initialize part of policy set by core

Message-id: <1409235879-1283-66-git-send-email-prarit@redhat.com>
Patchwork-id: 89379
O-Subject: [RHEL7.1 PATCH BZ 1134639 065/185] cpufreq: acpi-cpufreq: don't initialize part of policy set by core
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 2f75db781b88df2ee4bd62254efbb2768e3619bd
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Thu Oct 3 20:28:31 2013 +0530

    cpufreq: acpi-cpufreq: don't initialize part of policy set by core

    Many common initializations of struct policy are moved to core now and hence
    this driver doesn't need to do it. This patch removes such code.

    Most recent of those changes is to call ->get() in the core after calling
    ->init().

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/acpi-cpufreq.c b/drivers/cpufreq/acpi-cpufreq.c
index 5ce7aba..0d3589b 100644
--- a/drivers/cpufreq/acpi-cpufreq.c
+++ b/drivers/cpufreq/acpi-cpufreq.c
@@ -842,7 +842,6 @@ static int acpi_cpufreq_cpu_init(struct cpufreq_policy *policy)
   break;
  case ACPI_ADR_SPACE_FIXED_HARDWARE:
   acpi_cpufreq_driver.get = get_cur_freq_on_cpu;
-  policy->cur = get_cur_freq_on_cpu(cpu);
   break;
  default:
   break;
-- 
1.7.1